lib/membrane_file/source.ex

```diff
@@ -113,6 +139,32 @@ defmodule Membrane.File.Source do
     do_supply_demand(min(demand_size, size_to_read), redemand, state)
   end
 
+  defp do_supply_demand(to_supply_size, redemand, %{location: :stdin} = state) do
+    {buffer_actions, supplied_size} =
+      case IO.binread(to_supply_size) do
```

Comment:
Is `IO.binread(to_supply_size:: integer())` blocking until `to_supply_size` bytes of data are available or `:eof` is received?

Reply:
I'm not sure I understand, it returns on both receiving `to_supply_bytes` and encountering `:eof`

Reply:
I was just wondering if there is a possibility, that the `IO.binread/2` would return a number of bytes smaller than `to_supply_size`, but as far as I can see it is blocking untill the desired number of bytes is provided